### PR TITLE
ADM-365: add DeploymentFrequencySettings in metric step

### PR DESCRIPTION
**Pull request by LanxinWang**
## Summary
add DeploymentFrequencySettings in metric step

## After

image
image

## Note

_Null_

### CODE REVIEW

**Comment by guzhongren** [comment]
Why contain backend change(application.yaml)?

**Comment by LanxinWang** [comment]
> Why contain backend change(application.yaml)?

We pulled other branch by mistake, and now they have already been cleared up.

File: frontend/src/store.ts [hunk 1]
```diff
@@ -19,6 +20,7 @@ export const store = configureStore({
     jiraVerifyResponse: jiraVerifyResponseReducer,
     pipelineToolVerifyResponse: pipelineToolResponseReducer,
     sourceControlVerifyResponse: sourceControlVerifyResponseReducer,
+    pipelineMetricsSettings: pipelineMetricsSettingsReducer,
```
Comment (guzhongren) [inline frontend/src/store.ts:23]: Put it as a child of metrics reducer, ref to https://github.com/au-heartbeat/HeartBeat/blob/main/frontend/src/context/config/configSlice.tsx#L62

File: frontend/src/components/Metrics/MetricsStep/DeploymentFrequencySettings/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  return (
+    <>
+      <Divider>
+        <Title>Deployment frequency settings</Title>
+      </Divider>
+      {deploymentFrequencySettings.map((deploymentFrequencySetting, index) => (
+        <PipelineMetricSelection
+          key={uuidV4()}
```
Comment (guzhongren) [inline frontend/src/components/Metrics/MetricsStep/DeploymentFrequencySettings/index.tsx:26]: UUID as a key of the component is not recommended, coc, it take memory & time to generate it. 

Just use index, and keep [Keys Must Only Be Unique Among Siblings](https://rescript-lang.org/docs/react/latest/arrays-and-keys#keys-must-only-be-unique-among-siblings),

File: frontend/src/components/Metrics/MetricsStep/DeploymentFrequencySettings/style.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { basicButtonStyle } from '../../MetricsStepper/style'
+
+export const Divider = styled('div')({
+  padding: '0.4rem',
+  borderLeft: `0.4rem solid ${theme.main.backgroundColor}`,
+  margin: '1rem 0',
+})
+
+export const Title = styled('span')({
+  fontFamily: theme.typography.fontFamily,
```
Comment (guzhongren) [inline frontend/src/components/Metrics/MetricsStep/DeploymentFrequencySettings/style.tsx:13]: use default font family, and not specify in our app,
delete it

**Review by guzhongren — COMMENTED** [review]
Change with comments

**Pull request closed without merging**